src/pykeen/training/training_loop.py

```diff
@@ -673,3 +729,52 @@ def _free_graph_and_cache(self):
         self.model.regularizer.reset()
         # The cache of the previous run has to be freed to allow accurate memory availability estimates
         torch.cuda.empty_cache()
+
+    def save_state(self, path: str) -> None:
+        """Save the state of the training loop.
+
+        :param path:
+            Path of the file where to store the state in.
+        """
+        logger.debug("=> Saving checkpoint.")
+        torch.save(
+            {
+                'epoch': self._epoch,
```

Comment:
if you try running save_state() and you didn't set a checkpoint file, it will get an AttributeError for `self._epoch`

Reply:
Good catch 👍 That should be fixed now.